### ISSUE TITLE
Issues when `TF_VAR_environment` is set with nested remote modules

### ISSUE THREAD

**Issue by anish-muk** [terraform-linters/tflint#2412]
### Summary

I am working with a Terraform configuration that has a variable named 'environment', and I am setting the value via environment variable in a CI/CD configuration using `TF_VAR_environment`. It seems that having this particular variable set causes issues in TFLint when using the `--recursive` flag in a Terraform configuration that calls nested remote modules.

Setting the value of the `environment` via the TFLint `--var` flag and unsetting `TF_VAR_environment` works as a workaround, but creates consistency issues in our pipeline configuration. Changing the naming of the `environment` variable also suffices, but it seems like variable names should not impact linting behavior in this fashion.

Other variables set via `TF_VAR_*` work fine. 

### Command

tflint --recursive --minimum-failure-severity=warning

### Terraform Configuration

```terraform
terraform {
  required_version = ">= 1.7.0"
}

variable "environment" {
  type = string
}

module "myremotemodule" {
  source = "git::https://example.com/mymodule.git?ref=v1.2.0" # This module calls other remote modules internally
  
  environment = var.environment
}

output "myoutput" {
  value = var.environment
}
```

### TFLint Configuration

```HCL
plugin "azurerm" {
    enabled = true
    version = "0.29.0"
    source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}

plugin "terraform" {
  enabled = true
  preset  = "recommended"
}
```

### Output

```Shell
21:53:25 inspect_parallel.go:66: [DEBUG] Failed to run in azure/module; exit status 1; stdout=
│ Failed to run in azure/module; exit status 1

21:53:25 config.go:178: [INFO] Load config: .tflint.hcl
21:53:25 config.go:186: [INFO] file not found
21:53:25 config.go:193: [INFO] Load config: /home/runner/.tflint.hcl
21:53:25 config.go:335: [DEBUG] Config loaded
21:53:25 config.go:336: [DEBUG]   CallModuleType: local
21:53:25 config.go:337: [DEBUG]   CallModuleTypeSet: false
21:53:25 config.go:338: [DEBUG]   Force: false
21:53:25 config.go:339: [DEBUG]   ForceSet: false
21:53:25 config.go:340: [DEBUG]   DisabledByDefault: false
21:53:25 config.go:341: [DEBUG]   DisabledByDefaultSet: false
21:53:25 config.go:342: [DEBUG]   PluginDir: 
21:53:25 config.go:343: [DEBUG]   PluginDirSet: false
21:53:25 config.go:344: [DEBUG]   Format: 
21:53:25 config.go:345: [DEBUG]   FormatSet: false
21:53:25 config.go:346: [DEBUG]   Varfiles: 
21:53:25 config.go:347: [DEBUG]   Variables: 
21:53:25 config.go:348: [DEBUG]   Only: 
21:53:25 config.go:349: [DEBUG]   IgnoreModules:
21:53:25 config.go:353: [DEBUG]   Rules:
21:53:25 config.go:357: [DEBUG]   Plugins:
21:53:25 config.go:359: [DEBUG]     azurerm: enabled=true, version=0.29.0, source=github.com/terraform-linters/tflint-ruleset-azurerm
21:53:25 config.go:359: [DEBUG]     terraform: enabled=true, version=, source=
21:53:25 option.go:77: [DEBUG] CLI Options
21:53:25 option.go:78: [DEBUG]   CallModuleType: local
21:53:25 option.go:79: [DEBUG]   Force: true
21:53:25 option.go:80: [DEBUG]   Format: 
21:53:25 option.go:81: [DEBUG]   Varfiles: 
21:53:25 option.go:82: [DEBUG]   Variables: 
21:53:25 option.go:83: [DEBUG]   EnableRules: 
21:53:25 option.go:84: [DEBUG]   DisableRules: terraform_module_pinned_source
21:53:25 option.go:85: [DEBUG]   Only: 
21:53:25 option.go:86: [DEBUG]   EnablePlugins: 
21:53:25 option.go:87: [DEBUG]   IgnoreModules:
21:53:25 loader.go:39: [INFO] Initialize new loader
21:53:25 module_mgr.go:63: [INFO] Module manifest file found. Initializing...
21:53:25 loader.go:81: [INFO] Building the root module while calling local child modules...
21:53:25 runner.go:47: [INFO] Initialize new runner for root
21:53:25 input_value.go:67: [INFO] TF_VAR_* environment variable found: key=TF_VAR_environment
21:53:25 input_value.go:67: [INFO] TF_VAR_* environment variable found: key=TF_VAR_region
Failed to initialize a runner; <value for var.environment>:1,1-5: Variables not allowed; Variables may not be used here.:

Error: Variables not allowed

  on <value for var.environment> line 1:
  (source code not available)

Variables may not be used here.
```

### TFLint Version

0.59.1

### Terraform Version

1.13.1

### Operating System

- [ ] Linux
- [x] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
This was tricky to figure out without reproduction! 😉  

For what it's worth, recursive wasn't related. It's that TFLint was passing `TF_VAR_*` values to child modules. `--recursive` visits local modules as root modules. A child module is one referenced in a `module` block, whether local or remote, that is being evaluated as part of a rule. 

https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/calling-modules.md

There's nothing special about `environment` to TFLint, but there is to your private module. Somewhere in its dependency chain there's a `variable "environment" {}` block with a different `type`.

That said, this is a bug because environment variable should only affect the root module. Child modules receive variable values via explicit passing or defaults only.
